### PR TITLE
Fixes #3 Added Tests

### LINKED ISSUE #3: Add more tests
this is an evolutionary issue. and one could start by contributing the tests to those functions not covered in the test suite .
addition of the code coverage tools could also be done.

### PR DESCRIPTION
**Pull request by harshit98**
I have added some test for each component in `.spec.ts` file. Resolves issue no #3 . I haven't tested them because I would like someone to review it. I wanted to know whether I'm doing it right or not. Correct me, if I'm wrong somewhere :)

### CODE REVIEW

**Comment by Achint08** [comment]
Hi @harshit98 Try to avoid lint errors. This is a must to maintain project standards.

File: src/app/advancedsearch/advancedsearch.component.spec.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe('AdvancedsearchComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create an instance', () => {
     expect(component).toBeTruthy();
   });
-});
+});
```
Comment (Achint08) [inline src/app/advancedsearch/advancedsearch.component.spec.ts]: Always try to leave a extra line at the end of the file.
Do it for every other file.

**Comment by harshit98** [comment]
@Achint08 Hi. I'll make the changes as per your suggestions.

**Comment by harshit98** [comment]
@nikhilrayaprolu @mariobehling Okay. I have solved the issues which I was facing while running `ng lint` and `ng test`. Linting errors have been solved. Give me some time, I have to write some more test. I'll make some changes in travis.yml as well. Please don't close this PR as the work is still going on.

**Comment by mariobehling** [comment]
Hi, what's the status with this PR?

**Comment by harshit98** [comment]
@mariobehling Right now, we're having some issues regarding testing them. The tests which are written have not been checked due to some errors. You can close this PR. When the work will be completed, a new PR will be created. I have stopped working on this at the moment.

**Comment by nikhilrayaprolu** [comment]
@harshit98 I will look into your pull today and check with the errors .Hope we could merge tests early.
:)

**Pull request closed without merging**